### ISSUE TITLE
onShippingMethodSelected is not being called

### ISSUE THREAD

**Issue by elllot** [flutter-stripe/flutter_stripe#1156]
**Describe the bug**
`onShippingMethodSelected` is not being called following a new shipping method selection on the Apple Pay pay sheet. Additionally (perhaps as a consequence), the pay sheet gets stuck in the [processing state](https://drive.google.com/file/d/1eTvP1h7iU6TnNu3_dsNUY7owAl98JNuh/view?usp=share_link), which ends up in an exception after some time. The pay sheet is dismissed upon the failure. Following is the observed exception in the logs:
```
StripeException(error: LocalizedErrorMessage(code: FailureCode.Canceled, localizedMessage: The payment has been canceled, message: The payment has been canceled, stripeErrorCode: null, declineCode: null, type: null))
#0      ResultParser.parse (package:stripe_platform_interface/src/result_parser.dart:16:7)
#1      MethodChannelStripe.platformPayConfirmPaymentIntent (package:stripe_platform_interface/src/method_channel_stripe.dart:410:7)
<asynchronous suspension>
#2      _ApplePayScreenState._handlePayPress (package:stripe_example/screens/wallets/apple_pay_screen.dart:109:7)
<asynchronous suspension>
```

**To Reproduce**
Steps to reproduce the behavior (_note: this is observed on the [Flutter Stripe example app](https://github.com/flutter-stripe/flutter_stripe/tree/main/example)_):

1. Navigate to the Apple Pay screen by selecting the `Apple Pay` entry under the `Wallet` accordion ([screenshot](https://drive.google.com/file/d/1Wc2O24e1N9ZgwU8J05CqRfroF52EKqPs/view?usp=share_link)).
2. Tap the "Buy with apple pay" button.
3. Tap on shipping method to change shipping method.
4. Change shipping method.

**Expected behavior**
`onShippingMethodSelected` should be called with the new shipping method and the Pay sheet should switch back to the [state where user can proceed with the payment](https://drive.google.com/file/d/1C7dNJnBTrBhTfVwb7ouOELU8PVe7U4ND/view?usp=share_link).

**Smartphone / tablet**
 - Device: iPhone 14 Pro (emulator)
 - OS: iOS 16.0
 - Package version: 9.0.0
 - Flutter version 3.7.7

**Additional context**
I've tried both `flutter_stripe` versions `8.0.0` and `9.0.0` for the Example app.

Thanks for the help!

**Comment by elllot** [comment]
Hey team,

Were there any updates on this issue? I could possibly adjust the UX to avoid needing to update payment info based on `onShippingMethodSelected`, but still wanted to check if you guys have insights on the root cause. Thanks!

**Comment by remonh87** [comment]
solved with version 9.1.0
